pubtools/pulplib/_impl/model/repository/base.py

```diff
@@ -89,14 +91,15 @@ class Repository(PulpObject):
     """
 
     distributors = pulp_attrib(
-        default=attr.Factory(tuple),
-        type=tuple,
+        default=attr.Factory(FrozenList),
+        type=list,
```

Comment:
Why not "FrozenList"?
In docs there's:

> Please note that attrs doesn’t do anything with this metadata by itself. You can use it as part of your own code or for static type checking.

having list here makes it confusing

Reply:
The attrs docs aren't quite accurate, or at least, aren't telling the whole story.  These types do actually make it into the API as they are added as Python3 type annotations, where tools like pylint, mypy and sphinx look at them.  I expect them to end up in our docs eventually, though it's not quite working right now (https://github.com/release-engineering/pubtools-pulplib/issues/24).

For example, before this change the declared type of 'distributors' argument shows up in API as:

```
>>> inspect.signature(Repository.__init__).parameters['distributors'].annotation
<class 'tuple'>
```

After this change, it will be 'list'.  I do believe list is more appropriate than FrozenList here because:

- setting `type=FrozenList` is exposing that type as part of our public API, but I don't want to commit to returning instances of exactly that class
- given that it's used as type annotation on the constructor, the appropriate usage seems more like 'declare this as the type you *accept*' and not 'declare this as the type you *return*/*store*'.  So, type=list is fine as the constructor does indeed accept *any* type of list and doesn't have to be passed exactly a FrozenList.